modules/highgui/cmake/wayland.cmake

```diff
@@ -0,0 +1,24 @@
+macro(wayland_generate protocol_file output_file)
+    add_custom_command(OUTPUT ${output_file}.h
+            COMMAND ${WAYLAND_SCANNER_EXECUTABLE} client-header < ${protocol_file} > ${output_file}.h
+            DEPENDS ${protocol_file})
+    add_custom_command(OUTPUT ${output_file}.c
+            COMMAND ${WAYLAND_SCANNER_EXECUTABLE} private-code < ${protocol_file} > ${output_file}.c
+            DEPENDS ${protocol_file})
+    list(APPEND WAYLAND_PROTOCOL_SOURCES ${output_file}.h ${output_file}.c)
+endmacro()
+
+ocv_clear_vars(HAVE_WAYLAND WAYLAND_PROTOCOL_SOURCES)
+if (WITH_WAYLAND)
+    ocv_check_modules(WAYLAND wayland-client wayland-cursor wayland-egl xkbcommon)
+    ocv_check_modules(WAYLAND_PROTOCOLS wayland-protocols>=1.13)
+
+    pkg_get_variable(WAYLAND_PROTOCOLS_BASE wayland-protocols pkgdatadir)
```

Comment:
I think I have the proper solution now.  Only if protocols are found will it attempt to find the scanner, which is required.  If missing will cause build failure.  The pkg_get_variable command exists otherwise wayland-protocols.pc would never have been found.